src/Controllers/Version4/Responses/ProductResponse.php

```diff
@@ -78,29 +127,30 @@ public function prepare_response( $object, $context ) {
 			'reviews_allowed'       => $object->get_reviews_allowed( $context ),
 			'average_rating'        => $object->get_average_rating( $context ),
 			'rating_count'          => $object->get_rating_count(),
-			'related_ids'           => wp_parse_id_list( wc_get_related_products( $object->get_id() ) ),
 			'upsell_ids'            => wp_parse_id_list( $object->get_upsell_ids( $context ) ),
 			'cross_sell_ids'        => wp_parse_id_list( $object->get_cross_sell_ids( $context ) ),
 			'parent_id'             => $object->get_parent_id( $context ),
 			'purchase_note'         => $object->get_purchase_note( $context ),
 			'categories'            => $this->prepare_taxonomy_terms( $object ),
 			'tags'                  => $this->prepare_taxonomy_terms( $object, 'tag' ),
-			'images'                => $this->prepare_images( $object ),
-			'attributes'            => $this->prepare_attributes( $object ),
 			'default_attributes'    => $this->prepare_default_attributes( $object ),
 			'variations'            => array(),
 			'grouped_products'      => array(),
 			'menu_order'            => $object->get_menu_order( $context ),
-			'meta_data'             => $object->get_meta_data(),
 		);
```

Comment:
Won't most of these get handled by `fetch_fields_using_getters()`?

Reply:
Right now `fetch_fields_using_getters` only use methods defined in the controller. I am considering expanding it to define on object as well, in which case most of these will be handled automatically, but I am not sure if there are any security/backward compatibility implications for doing that.

Reply:
Why not define more methods like https://github.com/woocommerce/woocommerce-rest-api/pull/155/files#diff-f6ec5a652d294a7a96f0e0745fc6ddefR61 ?

I didn't see much of a performance improvement on the products endpoint in my local testing, but a huge (4x) improvement on orders.